docs/api.md

```diff
@@ -0,0 +1,14 @@
+# API Reference
+
+`adlfs.AzureBlobFileSystem` provides an interface for Azure Blob Storage.
```

Comment:
I'm not sure about the value of this page, at least `:members:` bit which just copies in stuff from upstream (warnings and all)

Reply:
Curious what others think but I'd lean toward referencing the core filesystem interfaces like is done in gcsfs: https://github.com/fsspec/gcsfs/blob/main/docs/source/api.rst